Lib/test/test_urlparse.py

```diff
@@ -890,13 +890,34 @@ def test_parse_qsl_max_num_fields(self):
             urllib.parse.parse_qs(';'.join(['a=a']*11), max_num_fields=10)
         urllib.parse.parse_qs('&'.join(['a=a']*10), max_num_fields=10)
 
+    def test_urlencode(self):
+        result = urllib.parse.urlencode({'a': 1, 'b': None})
+        # bpo-18857: urlencode of a None value uses the string 'None'
+        assert set(result.split('&')) == {'a=1', 'b=None'}
+
+    def test_urlencode_standalone_keys(self):
+        result = urllib.parse.urlencode(
+            query={'a': 1, 'b': None},
+            standalone_keys=True
+        )
+        assert set(result.split('&')) == {'a=1', 'b'}
+
     def test_urlencode_sequences(self):
         # Other tests incidentally urlencode things; test non-covered cases:
         # Sequence and object values.
-        result = urllib.parse.urlencode({'a': [1, 2], 'b': (3, 4, 5)}, True)
+        result = urllib.parse.urlencode({'a': [1, 2], 'b': (3, 4, 5), 'c': None}, True)
         # we cannot rely on ordering here
-        assert set(result.split('&')) == {'a=1', 'a=2', 'b=3', 'b=4', 'b=5'}
+        assert set(result.split('&')) == {'a=1', 'a=2', 'b=3', 'b=4', 'b=5', 'c=None'}
```

Comment:
```suggestion
        self.assertEqual(result, 'a=1&a=2&b=3&b=4&b=5&c=None')
```

Reply:
Thanks!  Your suggestion here (and the related assertion suggestions on previous lines) look like good robustness and readability improvements.

I've applied them locally and tests pass, so I'll batch them up in GitHub and hopefully should be able to apply them as a single commit retaining your authorship.  Here goes..